pytest_remotedata/disable_internet.py

```diff
@@ -51,22 +70,30 @@ def new_function(*args, **kwargs):
 
             host = args[0][0]
             addr_arg = 0
-            valid_hosts = ('localhost', '127.0.0.1')
+            valid_hosts = set(['localhost', '127.0.0.1'])
 
+        # Astropy + GitHub data
         if allow_astropy_data:
             for valid_host in ASTROPY_HOSTS:
-                valid_host_ip = socket.gethostbyname(valid_host)
-                valid_hosts += (valid_host, valid_host_ip)
+                valid_hosts = valid_hosts.union(_resolve_host_ips(valid_host))
+
+        # Only GitHub data
+        if allow_github_data and not allow_astropy_data:
```

Comment:
Looking at this again, perhaps this is better as `elif allow_github_data`, do you want me to change it now or do you just want to get this over with as-is?